cmd/argo-watcher/state/in_memory_state.go

```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/shini4i/argo-watcher/internal/models"
 )
 
+var taskIsObsolete = errors.New("task is obsolete")
```

Comment:
Not sure that moving errors into variables outside the function is a good idea. I don't see the benefit of doing it. 
If we need to re-use the error, it's better to re-organize the code.